server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -559,13 +560,23 @@ static void maybeSimulateAlreadyStartedShards(
     private void maybeLogAllocationExplainForUnassigned(
         DesiredBalance.ComputationFinishReason finishReason,
         RoutingNodes routingNodes,
-        RoutingAllocation routingAllocation
+        RoutingAllocation routingAllocation,
+        long inputIndex
     ) {
         if (allocationExplainLogger.isDebugEnabled()) {
+            final var clusterState = routingAllocation.getClusterState();
+            if (clusterState.metadata().nodeShutdowns().contains(clusterState.nodes().getLocalNodeId())) {
+                // Master is shutting down, the project is likely being deleted. Shards can be unassigned and no need to report
+                return;
+            }
```

Comment:
Logging on project deletion is one false postive that I noticed when enabling the logging in QA. 

The other false positve is during project creation, e.g. a search shard can be unassigned because search node has not joined yet. This is not addressed in this PR. I don't have a great solution other than some time-based delay, i.e. no logging in the first 5min after node start. 

I'd like to fix the issue on project deletion first since it is the more noisy case. I also added input index in the log which might be useful as a hint to decide whether to log or not for project creation if we observe some valid pattern in QA.